packages/cloud-functions/src/bigQuery.ts

```diff
@@ -15,6 +15,27 @@ export const trackEvent = async (table: string, row: any) => {
 
     await bigQuery.dataset(bigQueryDataset).table(table).insert(row)
   } catch (error) {
+    console.error(error)
+    console.error(`BigQuery error:`, JSON.stringify(error))
+    throw error
+  }
+}
+
+export const deleteDuplicates = async (table: string) => {
+  try {
+    if (!gcloudProject) {
+      throw new Error('No GCloud Project specified')
+    }
+
+    const sqlQuery = `
+      CREATE OR REPLACE TABLE ${bigQueryProjectId}.${bigQueryDataset}.${table}
```

Comment:
If it's creating it, will this work? Since it's doing a select from the same table. I'm not very familiar with this `CREATE OR REPLACE TABLE` statement.

Is it possible to add a constraint at a db level for whatever columns should be distinct instead?

Reply:
This won't work well for creating a new table from scratch but I don't think that's a problem because that's outside of the scope of this script. I am mainly doing this because some providers will send certain events multiple times.